aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -101,6 +102,11 @@ public String endpoint() {
     return properties().getOrDefault(S3_SIGNER_ENDPOINT, S3_SIGNER_DEFAULT_ENDPOINT);
   }
 
+  @Value.Lazy
+  public String endpointUri() {
```

Comment:
Why aren't we just changing the `endpoint()` behavior?  It looks like it's now only being used by `endpointUri()`, so can't we just update how we return the endpoint?